c2rust-analyze/src/type_desc.rs

```diff
@@ -52,3 +58,39 @@ pub fn perms_to_desc(perms: PermissionSet, flags: FlagSet) -> (Ownership, Quanti
 
     (own, qty)
 }
+
+pub fn perms_to_desc<'tcx>(
+    ptr_ty: Ty<'tcx>,
+    perms: PermissionSet,
+    flags: FlagSet,
+) -> TypeDesc<'tcx> {
+    let (own, qty) = perms_to_own_and_qty(perms, flags);
+
+    let pointee_ty = match *ptr_ty.kind() {
+        TyKind::Ref(_, ty, _) => ty,
+        TyKind::RawPtr(mt) => mt.ty,
+        TyKind::Adt(adt_def, substs) if adt_def.is_box() => substs.type_at(0),
+        // TODO: other ADTs, e.g. `Rc`
+        _ => panic!("expected a pointer type, but got {:?}", ptr_ty),
+    };
+
+    TypeDesc {
+        own,
+        qty,
+        pointee_ty,
+    }
+}
+
+pub fn local_perms_to_desc<'tcx>(
+    local_ty: Ty<'tcx>,
+    perms: PermissionSet,
+    flags: FlagSet,
+) -> TypeDesc<'tcx> {
+    let (own, qty) = perms_to_own_and_qty(perms, flags);
+    let pointee_ty = local_ty;
```

Comment:
this seems unnecessary and in my opinion would be clearer as the argument name changed to `pointee_ty` and function name changed to something like `pointer_perms_to_desc`

Reply:
This function isn't meant to be a general-purpose `TypeDesc` constructor, even if it happens to be implemented like one.  Added a doc comment to clarify: 7a2b68cd